qiskit/ignis/experiments/calibration/cal_table.py

```diff
@@ -143,7 +144,10 @@ def get_generator_kwargs(
         # convert (amp, phase) pair into complex value
         if use_complex_amplitude:
             if 'amp' in format_dict and 'phase' in format_dict:
-                format_dict['amp'] *= np.exp(1j * format_dict.pop('phase'))
+                if not isinstance(format_dict['amp'], ParameterExpression):
```

Comment:
Because it doesn't work now? We should avoid this change otherwise we cannot apply correct phase (`amp` is passed to schedule as parameter), or we might be able to add complex-valued amplitude in database, but this breaks data type of other entries.

You just need to checkout this branch to run the code.
https://github.com/Qiskit/qiskit-terra/pull/5192

Reply:
Not exactly. The reason for the change is the following: if the amplitude is a parameter then `phase` is still in the `format_dict` as `pop` was not called. When you pass the `format_dict` to a pulse function, e.g. Drag, it will fail with unexpected kwarg. Thus, if `amp` is a parameter expression I pop the phase. You could optimize the phase by treating amp as a complex parameter.

Reply:
I think pop is always called. This doesn't report correct amplitude value when the phase is nonzero. Could you please show some example code to reproduce this?

Reply:
what I expect here is ParameterExpression `amp = Parameter(amp) * exp(1j*Parameter(phase))`